rmp-serde/src/encode.rs

```diff
@@ -541,6 +600,265 @@ where
     }
 }
 
+impl<'a, W: Write + 'a> serde::Serializer for &mut ExtSerializer<'a, W> {
+    type Ok = ();
+    type Error = Error;
+
+    type SerializeSeq = Self;
+    type SerializeTuple = Self;
+    type SerializeTupleStruct = Self;
+    type SerializeTupleVariant = Self;
+    type SerializeMap = Self;
+    type SerializeStruct = Self;
+    type SerializeStructVariant = Self;
+
+    #[inline]
+    fn serialize_i8(self, value: i8) -> Result<Self::Ok, Self::Error> {
+        if self.tag.is_none() {
+            self.tag.replace(value);
+            Ok(())
+        } else {
+            Err(Error::InvalidDataModel("unexpected second i8"))
```

Comment:
For these error messages, could we briefly mention what _is_ expected?

Something like "expected ext bytes, found second i8" could make it much nicer to debug this invalid data model error. Even if all the other messages are "expected valid ext serialized data, found {seq, i16, etc.}" that'll let someone know that the problem is in their `_ExtStruct` implementation.